api/src/main/java/io/opentelemetry/trace/AttributeValue.java

```diff
@@ -221,4 +320,92 @@ public final Type getType() {
     @Override
     public abstract double getDoubleValue();
   }
+
+  @Immutable
+  @AutoValue
+  abstract static class AttributeValueStringArray extends AttributeValue {
+
+    AttributeValueStringArray() {}
+
+    static AttributeValue create(String... stringValues) {
+      return new AutoValue_AttributeValue_AttributeValueStringArray(Arrays.asList(stringValues));
+    }
+
+    @Override
+    public final Type getType() {
+      return Type.STRING_ARRAY;
+    }
+
+    @Override
+    public abstract List<String> getStringArrayValue();
+  }
+
+  @Immutable
+  @AutoValue
+  abstract static class AttributeValueBooleanArray extends AttributeValue {
+
+    AttributeValueBooleanArray() {}
+
+    static AttributeValue create(boolean... booleanValues) {
+      List<Boolean> values = new ArrayList<>(booleanValues.length);
```

Comment:
This will throw if `null` is passed, same for double and long below.
Attempts to set `null` (array) attributes should be silently ignored as decided in #765.

Reply:
Thanks for fixing the calls coming from `RecordEventsReadableSpan` in the SDK!
Since this is in the API, I'd add that null must not be passed to `AttributeValue`'s apidoc and probably annotate the params with `@NotNull`.
Here in this code you should still handle it gracefully in order to not crash if it's misused by returning an `AttributeValueBooleanArray` holding a value of null like `AttributeValueString.create(null)` does.

Reply:
I would argue to annotate the Javadoc with `@throws NullPointerException` as it is currently for the `AttributeValue.stringAttributeValue(String)` method. WDYT?

Reply:
Oh, that annotation is actually wrong since it doesn't throw --> #808

Regarding error handling, the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md) defines:
> API methods MUST NOT throw unhandled exceptions when used incorrectly by end users. The API and SDK SHOULD provide safe defaults for missing or invalid arguments. For instance, a name like `empty` may be used if the user passes in `null` as the span name argument during `Span` construction.

Reply:
Ugh. I don't think the SDK should be doing anything like that. That's definitely an exporter responsibility, not the SDK itself.

Reply:
Well the exporter could also drop null-strings 😉